app/src/main/java/org/openobservatory/ooniprobe/receiver/TestRunBroadRequestReceiver.java

```diff
@@ -0,0 +1,122 @@
+package org.openobservatory.ooniprobe.receiver;
+
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import com.google.common.collect.Lists;
+import com.google.common.math.Stats;
+
+import org.openobservatory.ooniprobe.common.PreferenceManager;
+import org.openobservatory.ooniprobe.common.service.RunTestService;
+import org.openobservatory.ooniprobe.test.TestAsyncTask;
+import org.openobservatory.ooniprobe.test.suite.AbstractSuite;
+
+import java.util.List;
+
+public class TestRunBroadRequestReceiver extends BroadcastReceiver implements ServiceConnection {
+    private final EventListener listener;
+    private final PreferenceManager preferenceManager;
+    public RunTestService service;
```

Comment:
I think the diff would feel a bit more encapsulated if we'd make this field private. (Of course to do this we would need to factor the `.service.task.interrupt()` call as I suggested above.)